java-example/src/main/java/com/devcycle/javaexample/MainActivity.java

```diff
@@ -0,0 +1,74 @@
+package com.devcycle.javaexample;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+
+import android.os.Bundle;
+import android.widget.Toast;
+
+import com.devcycle.sdk.android.api.DVCCallback;
+import com.devcycle.sdk.android.api.DVCClient;
+import com.devcycle.sdk.android.model.DVCEvent;
+import com.devcycle.sdk.android.model.DVCUser;
+import com.devcycle.sdk.android.model.Variable;
+import com.devcycle.sdk.android.util.LogLevel;
+
+import java.util.HashMap;
+import java.util.Objects;
+
+public class MainActivity extends AppCompatActivity {
+
+    Variable<String> variable = null;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        DVCClient client = DVCClient.builder()
+                .withContext(getApplicationContext())
+                .withUser(
+                        DVCUser.builder()
+                                .withUserId("nic_test")
+                                .build()
+                )
+                .withEnvironmentKey("add-mobile-key")
+                .withLogLevel(LogLevel.DEBUG)
+                .build();
+
+        variable = client.variable("activate-flag", "not activated");
```

Comment:
This should probably be in the `.onInitialized()` method or use the callback, or it will always have the default value here.

Reply:
The point is to show the variable Value change from when the client is initialized. I'll add some notes.